src/clone/Cloner.php

```diff
@@ -0,0 +1,250 @@
+				$command .= " --public-dir=$path";
+			}
+			$command .= $this->get_ssl_args( $source_site, $assoc_args );
+		}
+
+		if ( in_array( $site_details['site_type'], [ 'php', 'wp' ] ) ) {
+			if ( ! empty( $site_details['cache_nginx_browser'] ) ) {
+				$command .= " --cache";
+			}
+			if ( ! empty( $site_details['cache_host'] ) ) {
+				$command .= " --with-local-redis";
+			}
+			if ( ( ! empty( $site_details['db_host'] ) && 'php' === $site_details['site_type'] ) || 'wp' === $site_details['site_type'] ) {
+				if ( 'php' === $site_details['site_type'] ) {
+					$command .= " --with-db";
+				}
+			}
+			$command .= " --php=${site_details['php_version']}";
+		}
+
+		if ( 'wp' === $site_details['site_type'] ) {
+			if ( ! empty( $site_details['proxy_cache'] ) ) {
+				$command .= " --proxy-cache=on";
+			}
+			if ( ! empty( $site_details['app_sub_type'] ) && 'wp' !== $site_details['app_sub_type'] ) {
+				$command .= " --mu=${site_details['app_sub_type']}";
+			}
+			if ( ! empty( $site_details['app_admin_username'] ) ) {
+				$command .= " --admin-user=${site_details['app_admin_username']}";
+			}
+			if ( ! empty( $site_details['app_admin_email'] ) ) {
+				$command .= " --admin-email=${site_details['app_admin_email']}";
+			}
+			if ( ! empty( $site_details['app_admin_password'] ) ) {
+				$command .= " --admin-pass=${site_details['app_admin_password']}";
+			}
+			// TODO: vip, proxy-cache-max-time, proxy-cache-max-size
+		}
+
+		return $command;
+	}
+
+	public function create_site( Site $source_site, $assoc_args ): EE\ProcessRun {
+		EE::log( 'Creating site' );
+		EE::debug( 'Creating site "' . $this->name . '" on "' . $this->host . '"' );
+
+		$this->ensure_site_not_exists();
+		$new_site = $this->execute( $this->get_site_create_command( $source_site, $assoc_args ) );
+		$this->set_site_details();
+
+		return $new_site;
+	}
+
+	public function site_exists(): bool {
+		return 0 === $this->execute( 'ee site info ' . $this->name )->return_code;
+	}
+
+	public function ensure_site_exists(): void {
+		if ( ! $this->site_exists() ) {
+			EE::error( 'Unable to find \'' . $this->name . '\' on \'' . $this->host . '\'' );
+		}
+	}
+
+	public function ensure_site_not_exists(): void {
+		if ( $this->site_exists() ) {
+			EE::error( 'Site  \'' . $this->name . '\' already exists on \'' . $this->host . '\'' );
+		}
+	}
+
+	public function ssh_success(): bool {
+		return 0 === $this->execute( 'exit' )->return_code;
+	}
+
+	public function ensure_ssh_success(): void {
+		if ( ! $this->ssh_success() ) {
+			EE::error( 'Unable to SSH to ' . $this->host );
+		}
+	}
+
+	function set_site_details(): void {
+		$output  = $this->execute( 'ee site info ' . $this->name . ' --format=json' );
+		$details = json_decode( $output->stdout, true );
+		if ( empty( $details ) ) {
+			EE::error( 'Unable to get site info for site ' . $this->name . '. The output of command is: ' . $output->stdout . $output->stderr );
```

Comment:
Need to update error message in case site does not exist.
Output of command should be in debug log.

Reply:
1. Get `ee site list --format=json` from remote.
2. If the return code is non-zero, display: `Error fetching site list from remote server. Please try again and check logs.`
3. If the return code is zero and you get the list, just check if the site is there or not in the list. If it is not there, display: `Site ' . $this->name . ' does not exist on remote server.`
4. If the site is there on the remote server, but it is disabled, display message: `Site ' . $this->name . ' is disabled on remote server, please enable it to clone the site.`